example-shape-library/editorextensions/example-shape-library/src/extension.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  EditorClient,
+  Menu,
+  MenuLocation,
+  MenuType,
+  Viewport,
+} from "lucid-extension-sdk";
+
+const client = new EditorClient();
+const menu = new Menu(client);
+const viewport = new Viewport(client);
+
+client.registerAction("drawCustomShape", async () => {
+  const page = viewport.getCurrentPage();
+  if (!page) return;
+
+  const customShapeDef = await client.getCustomShapeDefinition(
+    "example-shape-library",
+    "Rectangle"
+  );
+  if (!customShapeDef) return;
+
+  const position = viewport.findAvailableSpace(
+    customShapeDef.boundingBox.h,
+    customShapeDef.boundingBox.w
+  );
+  page.addBlock({
+    ...customShapeDef,
+    ...position,
+  });
+});
+
+client.registerAction("drawSalesforceShape", async () => {
+  await client.loadBlockClasses(["SFACard"]);
+  const page = viewport.getCurrentPage();
+  if (!page) return;
+
+  const size = {
+    h: 80,
+    w: 320,
+  };
+
+  const position = viewport.findAvailableSpace(size.w, size.h);
+  page.addBlock({
+    className: "SFACard",
+    boundingBox: {
+      ...position,
+      h: 80,
+      w: 320,
+    },
+  });
+});
+
+client.registerAction("logBlockClass", () => {
+  const selection = viewport.getSelectedItems().pop();
+  if (!selection) return;
+
+  const page = viewport.getCurrentPage();
+  if (!page) return;
+
+  const blockWithSameId = page.blocks.get(selection.id);
+  console.log(blockWithSameId.getClassName());
+});
+
+client.registerAction("singleShapeSelected", () => {
```

Comment:
I think this is boderline fine but hovering over the 'Log Block class' sends a lot of "action took" to console log, sometimes making it hard to find the logged class name among the other logs.

https://github.com/lucidsoftware/sample-lucid-extensions/assets/81441553/304626f7-96f8-417b-a80a-04c32ba82a46